### ISSUE TITLE
[Feature request]: include context window utilization percentage in Messages API response

### ISSUE THREAD

**Issue by ArielSmoliar** [anthropics/anthropic-sdk-typescript#1064]
## Problem                                                        
                                
  The Messages API response includes `usage.input_tokens` and `usage.output_tokens`,                                                                                                     
  but not the model's context window size or the utilization ratio. To determine
  utilization, the dev must know the model's context window size out-of-band                                                                                                       
  and compute the ratio themselves.                                                                                                                                                      
                                                                                                                                                                                         
  This makes under-utilization invisible. In our case, we were chunking API calls at                                                                                                     
  600 log entries per call, using ~15% of the 200k context window, and didn't catch                                                                                                    
  it until we manually did the token budget math weeks later. The arbitrary chunking                                                                                                     
  introduced cross-call blindness and sequential latency for no reason.                                                                                                                  
                                                                                                                                                                                         
  ## Proposal                                                                                                                                                                            
                                                                                                                                                                                         
  Add `usage.context_window` and/or `usage.utilization` to the Messages API response:
                                      
  {                                                                                                                                                                                      
    "usage": {                        
      "input_tokens": 28000,                                                                                                                                                             
      "output_tokens": 1200,                                        
      "cache_creation_input_tokens": 1800,
      "cache_read_input_tokens": 600,                                                                                                                                                    
      "context_window": 200000,       
      "utilization": 0.146                                                                                                                                                               
    }                                                               
  }                             
                           
  When you see `"utilization": 0.146` in your logs, you immediately question why you're                                                                                                  
  splitting calls. When you see `"input_tokens": 28000`, you don't because the
  denominator is missing.                                                                                                                                                                
                                                                    
  ## Use case                                                                                                                                                                            
                                                                    
  Any application that chunks or batches input to fit the context window benefits from                                                                                                   
  knowing how much of the window each call actually uses. This is especially relevant
  for log analysis, document processing, and other structured-data                                                                                                        
  workloads where dev set conservative chunk sizes without verifying them                                                                                                         
  against the actual token budget.

**Comment by ArielSmoliar** [comment]
This would also enable a scheduling/orchestration layer to make real-time routing decisions. Right now the only utilization signal available in the Agent SDK is RateLimitEvent.utilization, which is a lagging indicator at the session level.

If the Messages API returned utilization per call, a scheduler could:                                                                                                                                              
* Route to a smaller/cheaper model when utilization is consistently low
* Batch multiple small calls into one when the window is underused                                                                                                                                                 
* Flag agents that are wasting context budget before costs compound

**Comment by dtmeadows-ant** [comment]
Hi @ArielSmoliar, the Models API now returns the context window, so this is computable without hardcoding anything:

```ts
const model = await client.models.retrieve('claude-sonnet-4-6');
const utilization = message.usage.input_tokens / model.max_input_tokens;
```

`max_input_tokens` and `max_tokens` are both on the model object, so one cached lookup per model covers it.
